server/config/config.go

```diff
@@ -315,7 +313,12 @@ func (c *ServerConfig) WALDir() string {
 func (c *ServerConfig) SnapDir() string { return filepath.Join(c.MemberDir(), "snap") }
 
 func (c *ServerConfig) ShouldDiscover() bool {
-	return c.DiscoveryURL != "" || len(c.DiscoveryCfg.Endpoints) > 0
+	for _, ep := range c.DiscoveryCfg.Endpoints {
+		if ep != "" {
```

Comment:
Why we would allow empty endpoint?

Reply:
`ShouldDiscover` returns true as long as there is at least one non-empty endpoint.

Reply:
updated. also added a unit test `TestDiscoveryCfg`